crypto/cipher_extra/e_chacha20poly1305.c

```diff
@@ -346,3 +382,317 @@ const EVP_AEAD *EVP_aead_chacha20_poly1305(void) {
 const EVP_AEAD *EVP_aead_xchacha20_poly1305(void) {
   return &aead_xchacha20_poly1305;
 }
+
+static int cipher_chacha20_poly1305_init_key(CIPHER_CHACHA_POLY_CTX *ctx,
+                                 const uint8_t user_key[CHACHA_KEY_LEN],
+                                 const uint8_t counter_nonce[CHACHA_CTR_IV_LEN])
+{
+  CIPHER_CHACHA_KEY *key = CC_KEY(ctx);
+  uint32_t i;
+  if (user_key) {
+    for (i = 0; i < (CHACHA_KEY_LEN / 4); i++) {
+      key->key[i] = *(((const uint32_t *) user_key) + i);
+    }
+  }
```

Comment:
This will most likely hit a endian issue. Can this use something like `CRYPTO_load_*`?

Reply:
I'll send the WIP docs, https://github.com/aws/aws-lc/pull/1318 will add CI for BE and help you test for it. The short answer is most algorithms (e.g. [ChaCha20](https://datatracker.ietf.org/doc/html/rfc7539#section-2.3) takes everything as little) specify the input/output is big/little, and historically all of our code assumed little, but swapping to the correct CRYPTO_load_u32_le will do the correct ([either swap or not](https://github.com/aws/aws-lc/blob/main/crypto/internal.h#L894-L898)) thing on big/little platforms. Without this you're copying bytes into a uint32 in the wrong order.